sqlalchemy_json/__init__.py

```diff
@@ -50,14 +50,28 @@ def coerce(cls, key, value):
             return NestedMutableList.coerce(key, value)
         return super(cls).coerce(key, value)
 
+class MutableListOrDict(Mutable):
+    """SQLAlchemy `mutable` extension with change tracking for a single-depth list or dict."""
+
+    @classmethod
+    def coerce(cls, key, value):
+        if value is None:
+            return value
+        if isinstance(value, cls):
+            return value
+        if isinstance(value, dict):
+            return MutableDict.coerce(key, value)
+        if isinstance(value, list):
+            return MutableList.coerce(key, value)
+        return super(cls).coerce(key, value)
 
 def mutable_json_type(dbtype=JSON, nested=False):
     """Type creator for (optionally nested) mutable JSON column types.
 
     The default backend data type is sqlalchemy.types.JSON, but can be set to
     any other type by providing the `dbtype` parameter.
     """
-    mutable_type = NestedMutable if nested else MutableDict
+    mutable_type = NestedMutable if nested else MutableListOrDict
```

Comment:
I'm not 100% sure on the name here, but that might be a bit of a wider concern, naming this `NestedJson` isn't possible as the name gets taken by the _column_ type later (which I think is more generally useful in exports), but getting it to better capture its meaning *and* then align it with the `Nested...` variant would be nice.

Suggestions are welcome if you have any.